### PR TITLE
feat(account-lib): STLX-663 From implementation for transfer builder

### PR DESCRIPTION
**Pull request by smaria95**
From implementation for transfer transactions

[STLX-663](https://bitgoinc.atlassian.net/browse/STLX-663)

### CODE REVIEW

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts [hunk 1]
```diff
@@ -217,6 +217,14 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
     if (this._fee === undefined) {
       throw new BuildTransactionError('Invalid transaction: missing fee');
     }
+    if (!this._fee.gasLimit) {
+      throw new BuildTransactionError('Invalid transaction: missing gas limit');
+    }
+    try {
+      this.validateValue(new BigNumber(this._fee.gasLimit, 16));
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/cspr/transactionBuilder.ts:224]: is `this._fee.gasLimit` set in base 16?

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts [hunk 2]
```diff
@@ -226,6 +234,7 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
     if (this._source === undefined) {
       throw new BuildTransactionError('Invalid transaction: missing source');
     }
+    this.validateAddress(this._source);
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/cspr/transactionBuilder.ts:237]: Please also change the definition for `validateAddress()` to use `isValidAddress` from './utils' instead

File: modules/account-lib/src/coin/cspr/transferBuilder.ts [hunk 3]
```diff
@@ -19,11 +25,17 @@ export class TransferBuilder extends TransactionBuilder {
 
   /** @inheritdoc */
   protected async buildImplementation(): Promise<Transaction> {
+    const extraArguments = new Map<string, CLValue>();
+    if (this._toAddress) {
+      // TODO(STLX-1691): We are sending the destination address as string until impediment STLX-1691 is fixed.
+      // After that we will change this to send an instance of PublicKey instead.
+      extraArguments.set(TRANSFER_TO_ADDRESS, CLValue.string(this._toAddress));
+    }
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/cspr/transferBuilder.ts:33]: 👍

File: modules/account-lib/src/coin/cspr/transferBuilder.ts [hunk 4]
```diff
@@ -33,7 +45,9 @@ export class TransferBuilder extends TransactionBuilder {
   initBuilder(tx: Transaction): void {
     super.initBuilder(tx);
     this.transaction.setTransactionType(TransactionType.Send);
-    // TODO(STLX-663): init to and amount
+    this.to(this.getTransferDestinationAddress(tx.casperTx.session));
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/cspr/transferBuilder.ts:48]: Please also change the definition for `to()` to use isValidAddress from './utils' instead

File: modules/account-lib/src/coin/cspr/transferBuilder.ts [hunk 5]
```diff
@@ -81,22 +95,64 @@ export class TransferBuilder extends TransactionBuilder {
...
   transferId(id: number): this {
     if (!isValidTransferId(id)) {
-      throw new InvalidParameterValueError('Invalid amount');
+      throw new InvalidParameterValueError('Invalid transfer id');
     }
     this._transferId = id;
     return this;
   }
 
+  private getTransferDestinationAddress(transferTx: ExecutableDeployItem): string {
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/cspr/transferBuilder.ts:104]: please add jsdoc

File: modules/account-lib/src/coin/cspr/transferBuilder.ts [hunk 6]
```diff
@@ -81,22 +95,64 @@ export class TransferBuilder extends TransactionBuilder {
...
+    const toAddress = transferTx.getArgByName(TRANSFER_TO_ADDRESS);
+
+    if (!toAddress || !toAddress.isString()) {
+      throw new InvalidTransactionError('Transfer does not have a destination address defined');
+    }
+
+    return toAddress.asString();
+  }
+
+  private getTransferAmount(transferTx: ExecutableDeployItem): string {
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/cspr/transferBuilder.ts:114]: add jsdoc

File: modules/account-lib/src/coin/cspr/transferBuilder.ts [hunk 7]
```diff
@@ -81,22 +95,64 @@ export class TransferBuilder extends TransactionBuilder {
...
   //endregion
 
   //region Validators
   validateMandatoryFields(): void {
-    if (this._toAddress === undefined) {
+    if (!this._toAddress) {
       throw new BuildTransactionError('Invalid transaction: missing to');
     }
-    if (this._amount === undefined) {
+    if (!isValidPublicKey(this._toAddress)) {
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/cspr/transferBuilder.ts:147]:
```suggestion
    if (!isValidAddress(this._toAddress)) {

```

**Review by CapnMigraine — CHANGES_REQUESTED** [review]

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts (hunk 1 above)
Comment (smaria95) [reply modules/account-lib/src/coin/cspr/transactionBuilder.ts:224]: No, it's a mistake to set the base here.
I'll remove it. Thanks @CapnMigraine !

**Review by CapnMigraine — APPROVED** [review]
nice work!

**Pull request merged**